internal/config/config.go

```diff
@@ -89,6 +89,65 @@ type Config struct {
 
 // }
 
+// defaultConfig
+func defaultConfig() Config {
```

Comment:
bug: callers can't distinguish success and error, which could lead to unexpected behaviour.

Reply:
He issue here is that this is called inside a `sync.Once`, should we panic if the config is incorrect?

Reply:
Resolved in f4783a7bbe02fc582b58b0080f45c7f9bf1b01e2